File: app/src/main/java/com/vsquad/projects/govorillo/ui/fragment/auth/TelNumberValidator.java
package com.vsquad.projects.govorillo.ui.fragment.auth;

public class TelNumberValidator {
    private StringBuilder valide_string;

    public String validate(final String s){
        for (int i = s.length()-1; i > s.length()-11; i--) {
            valide_string.append(s.split("")[i]);
        }
        valide_string.append("7+");
        valide_string.reverse();
        return valide_string.toString();
    }
}

File: app/src/main/java/com/vsquad/projects/govorillo/ui/fragment/auth/TelNumberValidator.kt
package com.vsquad.projects.govorillo.ui.fragment.auth

class TelNumberValidator {
    private lateinit var valide_string: String

    fun validate(s: String): String {
        valide_string = s.replace("[^0-9]".toRegex(), "").reversed()
        if(valide_string.length >= 10){
            valide_string = (valide_string.substring(0, 10) + "7+").reversed()
        }else{
            valide_string = ""
        }

        return valide_string
    }
}
